1.0.0 Release

KAllan357 opened the pull request:
:construction:

For this pull:
- [x] #37
- [x] #33 (If I understand the issue correctly)
- [x] #29 
- [x] #34 
- [x] #32 
- [x] #39 
- [x] Code Documentation
- [x] Readme updates
- [x] Flow Chart
- [x] Changlog
- [x] Testing

The reworked flow should also solve #23.

Commit:
remove deprecated code. remove circular dependency on nexus cookbook

Commit:
remove dependency on nexus cookbook

Commit:
default ssl_verify to true when working with a nexus server

Commit:
add artifact_version and artifact_location to provider class

version and location are two separate things, but when using Nexus,
they are usually combined into location.

be less strict on what it means to be from_nexus? and save the correct
values from Chef::Artifact.get_actual_version to the provider instance

Commit:
docs and a helper method

Commit:
extract a greater variety of packages

Commit:
use tar xf and combine most extensions

Commit:
tons of documentation for methods in the provider

Commit:
first pass at new deploy flow

Commit:
add a new fixture to test procs

Commit:
continue working on deploy flow

Commit:
delete_previous_versions and set @deploy to the value of force

Commit:
remove some logging code. flesh out proc_tests fixture

Commit:
logic changes in manifest_differences?. rename the previous_versions methods

Commit:
move delete_previous_versions into ruby_block

Commit:
rework manifest checking logic

Commit:
first pass on pre_seed action

Commit:
loads more documentation

ivey commented on providers/deploy.rb:
```diff
@@ -24,76 +25,92 @@
 action :deploy do
-  next unless new_resource.force or not deployed?
-
   delete_previous_versions(:keep => new_resource.keep)
```
Seems like this should happen at the end of the deploy, after we succeed. I know that's not in the PR, but I just noticed it.

KAllan357 replied:
Agreed. This was actually one of the things I almost wanted to add to a "1.1.0" milestone.

ivey commented:
Some minor comments on individual commits as I was reading through. Overall looks awesome.

Commit:
readme updates

Commit:
more documentation and a changelog

Commit:
update readme

Commit:
add new fixtures and fix bugs found in provider

Commit:
bug fix for a deleted manifest file. more fixture tests for removing files

Commit:
missing newlines on readme author page

Commit:
the restart proc should run whenever the current symlink changes

KAllan357 commented:
I added b8f712654c05d8576df0c9cc12b835516762206d to address an issue I believe I found.

As mentioned in #23, the restart proc is something that seems like it should run when you are switching between already installed versions. Without the above commit (or something like it), a deploy to another already-installed version won't call the restart proc unless the manifest has been altered.

So I think the solution is to add an additional OR clause to the guard around the restart proc - "is this configured run going to end up changing the current symlink away from what its pointing to right now?"

Commit:
extract a proc running method. improve logging

ivey commented:
:+1:

Commit:
update readme and run_proc method

Commit:
fixed typo in default_action

capoferro commented on providers/deploy.rb:
```diff
@@ -24,76 +27,96 @@
-    recipe_eval(&new_resource.before_extract) if new_resource.before_extract
+  run_proc :before_deploy
 
+  if deploy? || new_resource.force
```
I wonder if we should set @deploy to true if new_resource.force instead of adding new_resource.force to all the conditionals. It seems we never check force where we don't also check deploy? so if we have force ensure @deploy is true rather than treating it as an additional conditional, then we can avoid redundant code and potential for bugs in future.

ivey replied:
I like this idea

KAllan357 replied:
That is a good idea. I will make the change.

capoferro commented on fixtures/artifact_test/recipes/artifact_rollback_tests.rb:
```diff
@@ -0,0 +1,81 @@
+# Cookbook Name:: artifact_test
+# Recipe:: artifact_rollback_tests
+#
+# Copyright 2012, YOUR_COMPANY_NAME
```
Riot Games

capoferro commented on fixtures/artifact_test/recipes/pre_seed_tests.rb:
```diff
@@ -0,0 +1,30 @@
+# Cookbook Name:: artifact_test
+# Recipe:: pre_seed_tests
+#
+# Copyright 2012, YOUR_COMPANY_NAME
```
Riot Games

capoferro commented on fixtures/artifact_test/recipes/proc_tests.rb:
```diff
@@ -0,0 +1,190 @@
+# Cookbook Name:: artifact_test
+# Recipe:: proc_tests
+#
+# Copyright 2012, YOUR_COMPANY_NAME
```
Riot Games

capoferro commented on fixtures/artifact_test/recipes/manifest_differences_tests.rb:
```diff
@@ -0,0 +1,78 @@
+# Cookbook Name:: artifact_test
+# Recipe:: manifest_differences_tests
+#
+# Copyright 2012, YOUR_COMPANY_NAME
```
Riot Games

Commit:
code review cleanup

add 'Riot Games' to fixture files for company name
add kallan to authors for resources/deploy.rb
move the check on new_resource.force into the manifest_differences? method
fix bug with ruby_block in delete_previous_versions and LocalJumpError

Commit:
return true when force

Commit:
update readme

Commit:
update readme with rdocs
